### PR TITLE
Buttons - Tags Creation/Selection and Data Import/Export

### PR DESCRIPTION
**Pull request by laura-king**
Contains:

- Button & Dialog Functionality (Tag Selection, Tag Creation, File Import with status report download , Project Download)
- Bug fix - Approved Project status change
- Bug fix - Tag creation, now allows tags of same name for different projects
- Backend changes - Import endpoint, returns JSON with status instead of dictionary
- Fetch wrapper change  - allows other types(multipart) instead of JSON, returns Blob
- SLAC logo added in to footer

### CODE REVIEW

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { AnchorButton, Button, Dialog, DialogBody, DialogFooter, FormGroup, HTMLSelect, InputGroup, Label, NonIdealState, Spinner } from "@blueprintjs/core";
+import { AnchorButton, Button, Dialog, DialogBody, DialogFooter, FormGroup, HTMLSelect, InputGroup, FileInput, Label, NonIdealState, Spinner, Text } from "@blueprintjs/core";
 import { useEffect, useMemo, useState } from "react";
 import { formatToLiccoDateTime } from "../utils/date_utils";
 import { Fetch, JsonErrorMsg } from "../utils/fetching";
 import { sortString } from "../utils/sort_utils";
-import { ProjectApprovalHistory, ProjectInfo, projectTransformTimeIntoDates } from "./project_model";
+import { ProjectApprovalHistory, ProjectInfo, projectTransformTimeIntoDates, ImportResult } from "./project_model";
+//import { File } from "buffer";
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:7]: nit: unecessary comment

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 2]
```diff
@@ -396,6 +397,176 @@ export const HistoryOfProjectApprovalsDialog: React.FC<{ isOpen: boolean, onClos
...
+    const submit = () => {
+        setIsSubmitting(true);
+        const data = new FormData();
+        data.append("file", selectedFile);
+        data.append("name", selectedFile.value)
+
+
+        Fetch.post<ImportResult>(`/ws/projects/${project._id}/import/`, { body: data, headers: { "Content-Type": "MULTIPART" } })
+            .then((resp) => {
+                console.log(resp);
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:429]: unnecessary log

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 3]
```diff
@@ -396,6 +397,176 @@ export const HistoryOfProjectApprovalsDialog: React.FC<{ isOpen: boolean, onClos
...
+    onClose: () => void,
+    onSubmit: () => void
+}> = ({ isOpen, project, onClose, onSubmit }) => {
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [dialogError, setDialogError] = useState('');
+    const [selectedFile, setSelectedFile] = useState('');
+    const [importResult, setImportResult] = useState(String);
+    const [robustReport, setRobustReport] = useState(String);
+
+    useEffect(() => {
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:414]: not used, you can remove this

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 4]
```diff
@@ -396,6 +397,176 @@ export const HistoryOfProjectApprovalsDialog: React.FC<{ isOpen: boolean, onClos
...
+                        onInputChange={e => importFileChosen(e)}
+                    />
+                </FormGroup>
+                {dialogError ? <p className="error">ERROR: {dialogError}</p> : null}
+                {renderImportResult()}
+            </DialogBody>
+            <DialogFooter actions={
+                <>
+                    <Button onClick={(e) => onClose()}>Cancel</Button>
+                    <Button onClick={(e) => submit()} intent="primary" loading={isSubmitting} disabled={project.name === ""}>Submit File</Button>
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:502]: I suppose this button should be hidden or change to close once the file is imported and import result rendered? Right now the user can resubmit the import.

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 5]
```diff
@@ -396,6 +397,176 @@ export const HistoryOfProjectApprovalsDialog: React.FC<{ isOpen: boolean, onClos
...
+                a.href = url;
+                const now = new Date().toISOString();
+                a.download = `${project.name}_${now}.csv`;
+                a.click();
+
+                onSubmit();
+                setDialogError("");
+            })
+            .catch((e) => {
+                let err = e as JsonErrorMsg;
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:546]: you can get rid of this line, see other catch examples.

File: react/src/app/projects/projects_overview_dialogs.tsx [hunk 6]
```diff
@@ -396,6 +397,176 @@ export const HistoryOfProjectApprovalsDialog: React.FC<{ isOpen: boolean, onClos
...
+
+    const importFileChosen = (e: any) => {
+        if (!e) {
+            return;
+        }
+        setSelectedFile(e.target.files[0]);
+    }
+
+    const downloadReport = () => {
+        // TODO: robustReport, download with same way as export... bring out helper function
```
Comment (GreatDanton) [inline react/src/app/projects/projects_overview_dialogs.tsx:456]: missing download

**Review by GreatDanton — DISMISSED** [review]

**Comment by laura-king** [comment]
Sorry you reviewed the pr when it was still a draft (hence all the comments and unfinished functionality). Functionality is complete and ready for review now.

**Comment by laura-king** [comment]
@GreatDanton, mind giving this a quick review again before merge? Functionality is complete.

**Comment by GreatDanton** [comment]
> @GreatDanton, mind giving this a quick review again before merge? Functionality is complete.

It looks fine to me.

**Pull request merged**